MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -888,14 +873,12 @@ private bool IsSaverEnabledForItem(IMetadataSaver saver, BaseItem item, LibraryO
         {
             var results = await provider.GetSearchResults(searchInfo, cancellationToken).ConfigureAwait(false);
 
-            var list = results.ToList();
-
-            foreach (var item in list)
+            foreach (var item in results)
             {
                 item.SearchProviderName = provider.Name;
             }
 
-            return list;
+            return results;
```

Comment:
You're enumerating in the foreach loop and then potentially again outside this function. Why?

Reply:
Again this was done before my changes. If you would like me to take a look at changing that I can but it was a bit outside what I had in mind.

Reply:
Fixed as best I could. I ended up adding back a `ToList` call because multiple enumerations were needed in some places.